sdk/keyvault/keyvault-keys/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -180,4 +187,32 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
       );
     }
   }
+
+  /**
+   * Parses an WWW-Authenticate response.
+   * This transforms a string value like:
+   * `Bearer authorization="some_authorization", resource="https://some.url"`
+   * into an object like:
+   * `{ authorization: "some_authorization", resource: "https://some.url" }`
+   * @param wwwAuthenticate string value in the WWW-Authenticate header
+   */
+  public parseWWWAuthenticate(wwwAuthenticate: string): ParsedWWWAuthenticate {
+    // First we split the string by either `, ` or ` `.
+    const parts = wwwAuthenticate.split(/,* /);
```

Comment:
Shouldn't that be `/(, )*/`? Could be multiple consecutive whitespace characters.

Reply:
I think it's better with `/,* */`, since we can have multiple spaces, but probably not multiple spaces preceded with a comma!

Thanks for the feedback though! Super appreciated 🌞

Reply:
Yeah, that's better. I wasn't thinking about them in sequence, but is obvious in hindsight. Decades of regex, but it'll still make fools of us all!

Reply:
Yes, regexes are so weird!